### PR TITLE
feat: [CDS-35104]: Added test case for CDS-25549 and removed duplication of code

### PR DESCRIPTION
**Pull request by goyalsonali**
Summary:
Jira Links:

https://harness.atlassian.net/browse/CDS-35104

 Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by goyalsonali** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/__tests__/__snapshots__/JiraCreate.test.tsx.snap [hunk 1]
```diff
@@ -690,6 +690,33 @@ exports[`Jira Create fetch projects show error if failed to fetch projects 1`] =
...
 
+exports[`Jira Create tests Basic snapshot - deploymentform mode with fields as runtime input: jira-create-deploymentform-fields-runtime 1`] = `
+<div>
+  <div
+    class="OverlaySpinner--overlaySpinner"
+  >
+    <p
+      class="StyledProps--font StyledProps--main"
+    >
+      Errors
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/__tests__/__snapshots__/JiraCreate.test.tsx.snap:701]: I think your test should render component with some fields properly. It is just rendering Errors here. Please check mocking of data and APIs properly.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/__tests__/__snapshots__/JiraCreate.test.tsx.snap (hunk 1 above)
Comment (goyalsonali) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/__tests__/__snapshots__/JiraCreate.test.tsx.snap:701]: Okay

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraFieldsRenderer.tsx [hunk 2]
```diff
@@ -38,6 +38,17 @@ function GetMappedFieldComponent({
   index,
   jiraContextType
 }: MappedComponentInterface) {
+  const multiType =
```
Comment (chiragp1212) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraFieldsRenderer.tsx:41]: Rename to more `meaningful`

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]
We need to change test case, so that it actually automate changes which are done as part of  CDS-25549 ticket.

**Comment by goyalsonali** [comment]
trigger rebase

**Comment by goyalsonali** [comment]
trigger rebase

**Comment by goyalsonali** [comment]
trigger cypress

**Review by chiragp1212 — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**